doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,451 @@
+
+Exposing policies to the end hosts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The path combinator finds the most recent FABRID map per AS among the received segments and subsequently uses this map to find the FABRID
+policies that are available for each interface pair of hops.
+This results in a set of PolicyIdentifiers per hop, which can then be used by the application, such as with the usage of a
+specific 'sequence' parameter which incorporates the policies.
+Once the application has decided which policies to use, it can craft a FABRID HBH extension and include this as an option when sending
+the packet.
+
+DRKey
+^^^^^^
+
+FABRID uses DRKey for computing the Encrypted Policy Indices, the FABRID Hop Validation Fields and the Path Validator.
+The routers use the fast key derivation side, whereas the endhosts will use the slow side.
+
+Configuration
+--------------
+
+Control service
+^^^^^^^^^^^^^^^^^^
+
+To be able to use DRKey, one has to configure the control service setting "drkey.level1_db" and "drkey.secret_value_db".
+Additionally, since the border routers will fetch the secret value from the control service, the control service also has to
+add the internal IP address of all border routers of the local AS to the DRKey delegation list for FABRID.
+
+This could look like this::
+
+    [drkey.level1_db]
+    connection = "gen-cache/cs1-ff00_0_110-1.drkey-level1.db"
+
+    [drkey.secret_value_db]
+    connection = "gen-cache/cs1-ff00_0_110-1.drkey-secret.db"
+
+    [drkey.delegation]
+    FABRID = [ "fd00:f00d:cafe::7f00:11", "fd00:f00d:cafe::7f00:12", "fd00:f00d:cafe::7f00:13"]
+
+The FABRID policies are configured in the control service.
+
+Border router
+^^^^^^^^^^^^^^^
+
+For a router to query the DRKey secret value from the control service, once has to enable this.
+
+This could look like this::
+
+    [router]
+    use_drkey = true
+
+Considerations for future work
+--------------------------------
+
+SCMP response
+^^^^^^^^^^^^^^^
+
+With the current implementation, the sending endhost is not being informed when his packet gets dropped due to a FABRID error.
+In the future the border routers we might think about whether to enable routers to send an SCMP response if they encounter
+an error when processing FABRID which might help the sending endhost in figuring out why his packet does not arrive at its destination.
+However, we have to be careful to not introduce additional attack vectors, e.g. for volumetric denial-of-service or reflection attacks.
+
+EPIC-HP as extension with Identifier option
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We could create a new HBH extension for EPIC hidden-path, which uses the Identifier option, instead of the
+current design which uses a SCION path-type.
+This allows the use of EPIC HP also in a incremental deployment like we have with FABRID.
+And additionally, we could also use FABRID together with EPIC HP.
+
+.. _RAINBOW:
+
+The RAINBOW system
```

Comment:
Add a URL for RAINBOW

Reply:
I don't have a url for RAINBOW.

Reply:
Indeed, the paper is still under submission.
I would suggest we add an URL once it is published.